TradeItIosTicketSDK2/TradeItDeviceManager.swift

```diff
@@ -0,0 +1,69 @@
+import LocalAuthentication
+
+class TradeItDeviceManager {
+    
+    func authenticateUserWithTouchId(onSuccess onSuccess: () -> Void, onFailure: () -> Void) {
+        var error: NSError?
+        let myLocalizedReasonString = "Authentication is required"
+        
+        let context = LAContext()
+        
+        guard context.canEvaluatePolicy(LAPolicy.DeviceOwnerAuthentication, error: &error) else {
+            print("touch id not available on this device: \(error)")
+            onSuccess()
+            return
+        }
+        
+        context.evaluatePolicy(LAPolicy.DeviceOwnerAuthentication,
+                               localizedReason: myLocalizedReasonString,
+                               reply: { (success: Bool, evaluationError: NSError?) -> Void in
+                                
+                                if success {
+                                    print("authenticateUserWithTouchId succeeded")
+                                    onSuccess()
+                                }
+                                else {
+                                    guard let error = evaluationError else {
+                                        print("authenticateUserWithTouchId failed")
+                                        onFailure()
+                                        return
+                                    }
+                                    
+                                    switch error.code {
```

Comment:
Yeah, I agree. We should just leave them where they currently are before they hit Trade or whatever.